### PR TITLE
View person alerts

### PR DESCRIPTION
**Pull request by hortha**
### Context

The TRS Console needs to be able to show the alerts for a person.

### Changes proposed in this pull request

Add a new page at /persons/{personId}/alerts following the designs at in Figma.

Within the alerts tab, show a Current alerts section. (A current Alert is one where the Status is Active and the End Date is null). If there none, show ‘No current alerts'. Alerts should be sorted in ascending Start date order. For each alert, show a summary card where the card title is the sanction’s dfeta_name. Add a ‘Close’ link to a stub page /alerts/{alertId}/close. Include a Start date row and a Details row. The ‘Add an alert’ button should link to /persons/{personId}/alerts/new (a stub page).

If there any previous alerts (a previous alert is one with a non-`null` End date or where the Status is Inactive) show an Previous alerts section with a table listing all the previous alerts, sorted by ascending start date. Each row should link to a stub page /alerts/{alertId}. The Status column should be ‘Closed’ if the End date is non-`null` or ‘Inactive’ if the Status is Inactive.

If the person with ID {personId} does not exist, return a 404. (This should probably be moved into a filter attribute for reusability).

### Guidance to review

CRM query + UI + tests.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[View person alerts](https://trello.com/c/VGKBJtbN/1226-view-person-alerts)

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    Layout = "_Layout";
+    var selectedTab = (PersonSubNavigationTab)ViewBag.SelectedTab;
+    var personId = (Guid)ViewBag.PersonId;
+    var search = (string)ViewBag.Search;
+    var sortBy = (ContactSearchSortByOption)ViewBag.SortBy;
+    var pageNumber = (int)ViewBag.PageNumber;
+}
+
+@section BeforeContent {
+    <govuk-back-link href="@LinkGenerator.Persons(search, sortBy, pageNumber)" />
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml:13]: Do we want this in the layout - isn't this back link page specific?

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@using TeachingRecordSystem.Core.Dqt.Models;
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml:1]: Could we move this into `Persons/Layout.cshtml`?

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs [hunk 3]
```diff
@@ -96,7 +96,15 @@
...
 builder.Services
-    .AddRazorPages()
+    .AddRazorPages(options =>
+    {
+        options.Conventions.AddFolderApplicationModelConvention(
+                "/Persons/PersonDetail",
+                model =>
+                {
+                    model.Filters.Add(new CheckPersonExistsFilter());                    
+                });
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs]:
```suggestion
            "/Persons/PersonDetail",
            model =>
            {
                model.Filters.Add(new CheckPersonExistsFilter());                    
            });
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+        var allAlerts = sanctions!.Select(MapSanction);
+
+        CurrentAlerts = allAlerts
+            .Where(alert => alert.Status == AlertStatus.Active)
+            .OrderBy(a => a.StartDate)
+            .ToArray();
+
+        PreviousAlerts = allAlerts
+            .Where(alert => alert.Status != AlertStatus.Active)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs:61]:
```suggestion
            .Except(CurrentAlerts)
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        {
+            if (sanction.Sanction.dfeta_EndDate is null)
+            {
+                alertStatus = AlertStatus.Active;
+            }
+            else
+            {
+                alertStatus = AlertStatus.Closed;
+            }
+        }
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs:81]:
```suggestion
        var alertStatus = sanction.Sanction.StateCode == dfeta_sanctionState.Inactive ? AlertStatus.Inactive :
            sanction.Sanction.dfeta_EndDate is null ? AlertStatus.Active :
            AlertStatus.Closed;
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml:13]: They all go back to the results screen

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml (hunk 2 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/PersonLayout.cshtml]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs (hunk 3 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs (hunk 5 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs (hunk 4 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Alerts.cshtml.cs]: Done

**Review by gunndabad — APPROVED** [review]

**Pull request merged**